pkg/pillar/scripts/device-steps.sh

```diff
@@ -501,15 +501,23 @@ echo "$(date -Ins -u) Done starting EVE version: $(cat /run/eve-release)"
 # If there is a USB stick inserted and debug.enable.usb is set, we periodically
 # check for any usb.json with DevicePortConfig, deposit our identity,
 # and dump any diag information
+(while true; do
```

Comment:
if you run in background, this subshell will not get killed when the script exits... I know that's not a big deal since the script already runs on a main loop, but I would do something like this:

```sh
#/bin/sh

func1() {
	while true; do
		echo func1
		sleep 4
	done
}

func2() {
	while true; do
		echo func2
		sleep 2
	done
}

func1 &
func2 &
trap "kill -9 -$$" SIGINT SIGTERM EXIT
sleep inf
```

if the script exits trap will run kill and send -9 to the process group, killing background processes...

Reply:
That's correct, but the script is not supposed to quit, and everything it starts should run forever. So I do not think we need anything like that.

Anyway, I rewrote the whole logic. The problem is that we never reach the final loop, because we always stuck even not in the USB, but in doing `selfRegister`.